### PR TITLE
feat(python): Introduce Initial Boilerplate for Python V2 Codegen

### PR DESCRIPTION
**Pull request by noanflaherty**
Introduces the initial package boilerplate for a new python v2 codegen using ast patterns.

## Summary by CodeRabbit

- **New Features**
	- Introduced new configuration files for improved package management and formatting.
	- Added an abstract class for enhanced AST handling.
	- Implemented a new representation within the AST framework, including methods for string representation and writing operations.
	- Added a new function for creating instances of the new class.

- **Tests**
	- Added test coverage for the new class functionality.

### CODE REVIEW

File: generators/pythonv2/codegen/src/ast/Class.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { Writer } from "./core/Writer";
+
+export declare namespace Class {
+    interface Args {
+        /* The name of the Python class */
+        name: string;
+    }
+}
+
+export class Class extends AstNode {
```
Comment (noanflaherty) [inline generators/pythonv2/codegen/src/ast/Class.ts:11]: This serves as an initial stub that'll have a lot more built out in future PRs.

File: generators/pythonv2/codegen/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "@fern-api/pythonv2-codegen",
```
Comment (noanflaherty) [inline generators/pythonv2/codegen/package.json:2]: This is petty much a copy/paste, but replaces `csharp` with `pythonv2`.

File: generators/pythonv2/codegen/src/ast/__test__/__snapshots__/Class.test.ts.snap [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
+
+exports[`class > basic 1`] = `"class Car:"`;
```
Comment (noanflaherty) [inline generators/pythonv2/codegen/src/ast/__test__/__snapshots__/Class.test.ts.snap:3]: This test will be updated to include more down the road.

File: generators/pythonv2/codegen/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "../../../shared/tsconfig.shared.json",
```
Comment (noanflaherty) [inline generators/pythonv2/codegen/tsconfig.json:2]: Copy/paste

File: generators/pythonv2/codegen/.depcheckrc.json
Comment (dvargas92495) [inline generators/pythonv2/codegen/.depcheckrc.json:1]: [q] what's depcheck?

**Review by dvargas92495 — APPROVED** [review]

File: generators/pythonv2/codegen/package.json (hunk 2 above)
Comment (dsinghvi) [reply generators/pythonv2/codegen/package.json:2]: nice!

File: generators/pythonv2/codegen/.depcheckrc.json
Comment (dsinghvi) [reply generators/pythonv2/codegen/.depcheckrc.json:1]: makes sure that you have no unused dependencies

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**